packages/api/cms-api/src/page-tree/page-tree.service.ts

```diff
@@ -504,6 +519,37 @@ export class PageTreeService {
                 return qb.getResultList();
             },
 
+            async getPaginatedNodes(options = {}, where = {}): Promise<[PageTreeNodeInterface[], number]> {
+                const qb = pageTreeNodeRepository.createQueryBuilder().where({
+                    visibility: visibilityFilter,
+                    ...where,
+                });
+
+                if (options.category) {
+                    qb.andWhere({ category: options.category });
+                }
+
+                if (options.scope) {
+                    qb.andWhere({ scope: options.scope });
+                }
+
+                if (options.documentType) {
+                    qb.andWhere({ documentType: options.documentType });
+                }
+
+                if (options.sort) {
+                    options.sort.map((sortItem) => {
+                        qb.orderBy({ [`${sortItem.field}`]: sortItem.direction });
+                    });
+                }
+
+                if (options.limit !== undefined && options.offset !== undefined) {
+                    qb.limit(options.limit, options.offset);
+                }
+
+                return [await qb.getResultList(), await qb.getCount()];
```

Comment:
Does this perform the same query twice?

Reply:
yes, but i could not find a better solution for this

Reply:
this requires always 2 queries (and no, it's not the same query) - even if using findAndCount